src/test/java/htsjdk/io/HtsPathUnitTest.java

```diff
@@ -649,4 +664,52 @@ private String getLocalFileAsURIPathString(final Path localPath) {
         return localPath.toUri().normalize().getPath();
     }
 
+    @DataProvider
+    public Object[][] getNonProblematic() {
+        return new Object[][]{
+                // URI is unencoded but no problems with a scheme
+                {"file/ name-sare/ :wierd-"},
+                {"hello there"},
+
+                // file scheme is deliberately ignored here since it's handled specially later
+                {"file://unencoded file names/ are/ok!"},
+
+                //these aren't invalid URIs so they should never be tested by this method, they'll pass it though
+
+                {"eep/ee:e:e:ep"},
+                {"file:///"},
+        };
+    }
+    @Test(dataProvider = "getNonProblematic")
+    public void testAssertNoProblematicScheme(String path){
+        assertNoProblematicScheme(path, null);
+    }
+
+    @DataProvider
+    public Object[][] getProblematic(){
+        return new Object[][]{
+
+                //This is the primary use case, to detect unencoded uris that were intended to be encoded
+                {"http://forgot.com/to encode"},
+                {"ftp://server.com/file name.txt"},
```

Comment:
So if I understand this correctly, `assertNoProblematicScheme()` only rejects these because they have a non-file scheme, and doesn't actually check the encoding at all, and the assumption is that we've already triggered a URI parsing exception? Might want to add a comment here explaining that

Reply:
yeah, it's stuck into the catch block so it's already guarenteed to have hit the exception, good idea to comment.  

honestly, it should probably be restructured so it's in the catch block but I wanted to change as little as possible